### PR TITLE
Allow spaces in namespace

### PR DESCRIPTION
**Pull request by kyle-dorman**

### CODE REVIEW

File: src/main/scala/com/gilt/gfc/aws/cloudwatch/periodic/metric/aggregator/CloudWatchMetricDataAggregatorBuilder.scala [hunk 1]
```diff
@@ -157,8 +157,8 @@ object CloudWatchMetricDataAggregatorBuilder
   private val DimNameMaxStrLen = 250
   private val GenericMaxStrLen = 256
 
-  private val DimValueAllowedChars = """[0-9A-Za-z.\-_/#:*]"""
-  private val GenericAllowedChars  = """[0-9A-Za-z.\-_/#:]"""
+  private val DimValueAllowedChars = """[0-9A-Za-z\w.\-_/#:*]"""
+  private val GenericAllowedChars  = """[0-9A-Za-z\w.\-_/#:]"""
```
Comment (andreyk0) [inline src/main/scala/com/gilt/gfc/aws/cloudwatch/periodic/metric/aggregator/CloudWatchMetricDataAggregatorBuilder.scala]: Do they really allow *all* white space or just ' ' characters only?

Comment (gheine) [inline src/main/scala/com/gilt/gfc/aws/cloudwatch/periodic/metric/aggregator/CloudWatchMetricDataAggregatorBuilder.scala]: Probably just space (ASCII 32)

File: src/main/scala/com/gilt/gfc/aws/cloudwatch/periodic/metric/aggregator/CloudWatchMetricDataAggregatorBuilder.scala [hunk 2]
```diff
@@ -95,15 +95,21 @@ case class CloudWatchMetricDataAggregatorBuilder private[metric] (
     this.copy(metricDimensions = this.metricDimensions :+ ds)
   }
 
+  /** namespace used when publishing. Exposed for testing */
+  def sanitizedNamespace: Option[String] = metricNamespace.map(_.limit())
```
Comment (andreyk0) [inline src/main/scala/com/gilt/gfc/aws/cloudwatch/periodic/metric/aggregator/CloudWatchMetricDataAggregatorBuilder.scala]: If it's just for testing you could try to make it 'package private', should work the same (assuming your test is in the same package) but will avoid polluting public API.

**Comment by andreyk0** [comment]
👍  thanks!

**Pull request merged**